jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -89,10 +110,34 @@ object XGBoost extends Serializable {
     } else {
       throw new IllegalArgumentException(
         s"Encountered a partition with $nUndefined NaN base margin values. " +
-            "If you want to specify base margin, ensure all values are non-NaN.")
+          "If you want to specify base margin, ensure all values are non-NaN.")
     }
   }
 
+  /**
+   * Create an RDD which is responsible for actually executing the training stage. Each RDD
+   * partition is converted to an XGBoost `DMatrix`, and a gradient boosted decision tree model
+   * is trained by calling into the native XGBoost code through the Java Native Interface.
+   *
+   * Each RDD partition is essentially a long-running task that repeatedly performs boosting
+   * iterations by calling into the native XGBoost library and communicates split statistics
+   * through the [[Rabit]] all-reduce framework instead of performing aggregations directly
+   * within Spark. Training is executed in a single Spark job, and can be triggered by invoking
+   * some action on the returned RDD - e.g. a call to `foreachPartition`.
+   *
+   * @param data the training set represented as an RDD
+   * @param params Hashmap containing the configuration entries
```

Comment:
I think the `@param` spec does not add much value, as the method is private and most of the arguments are already documented just above in the public wrappers.

Also, not sure about duplicating the type in the description of the param.

Reply:
agree with @superbobry